review/internal/api/api.go

```diff
@@ -2,80 +2,38 @@ package api
 
 import (
 	"context"
-	"coupon_service/internal/service/entity"
-	"fmt"
-	"log"
-	"net/http"
-	"time"
 
 	"github.com/gin-gonic/gin"
+
+	"coupon_service/internal/service/entity"
 )
 
 type Service interface {
-	ApplyCoupon(entity.Basket, string) (*entity.Basket, error)
-	CreateCoupon(int, string, int) any
-	GetCoupons([]string) ([]entity.Coupon, error)
-}
-
-type Config struct {
-	Host string
-	Port int
+	ApplyCoupon(context.Context, int, int, string) (entity.Basket, error)
```

Comment:
I passed context as a first argument across the handlers for future purposes (e.g. new storage using network interaction)